src/engine/outfit.ts

```diff
@@ -93,9 +94,11 @@ function optimisticCandle(): Familiar {
 }
 
 function melodramedary(): Familiar {
-  return have($familiar`Melodramedary`) &&
+  return (have($familiar`Melodramedary`) &&
     camelFightsLeft() >= Math.ceil((100 - get("camelSpit")) / 3.0) &&
-    get("camelSpit") < 100
+    get("camelSpit") < 100) || (have($familiar`Melodramedary`) &&
+    camelFightsLeft() >= Math.ceil((100 - get("camelSpit")) / 4.0) &&
+    get("camelSpit") < 100 && have($item`dromedary drinking helmet`))
```

Comment:
Both sides of the `A || B` have `get("camelSpit") < 100`, so it should be brought out of the or statement.

I also do not agree with this in principle. The division by 4.0 is an expectation, but you are not guaranteed to get spit by that number of turns - if it turns out that you're unlucky it could turn out to be extremely inefficient. I don't think there is any other logic in the script that forgoes a guaranteed strat for a probabilistic one